### PR TITLE
DBW node resets braking LPF when going from manual braking to acceleration, to prevent slow speed up from stop.

### PR DESCRIPTION
**Pull request by dinoboy197**
Per [Slack conversation](https://carnd.slack.com/archives/G8SCYKNSZ/p1528825626000228).

This resets the DBW brake low pass filter when switching to acceleration after manual braking. This should allow the vehicle to accelerate immediately, rather than waiting for the filtered braking value to lower through the lpf for several cycles.

### CODE REVIEW

**Comment by dinoboy197** [comment]
I'm on the road and don't have enough battery power to get the simulator up and running to do exhaustive tests, but I can test later on tonight.

**Comment by teeekay** [comment]
This works when the order of the elif blocks on lines 105 to 125 of twist_controller_mod.py is reversed.

**Comment by edufford** [comment]
As @teeekay suggested, switching the elif condition order for PID braking and accel control give the accel thresholds higher priority to clear the manual brake filter in the near-zero speed area transition, where the tiny speed values cause some ambiguity in the mode selection.

Originally I think PID braking was given higher priority than accel on principle to ensure braking was maintained during a slow down, but that was before we had the `is_decelerating` flag so maybe it's ok to switch the order for this issue.

An alternate way could be to guard the error threshold to prevent it from shrinking to near zero as the velocity approaches zero, but that would be more complicated and would need more side-effect checking for all the mode transitions so maybe not worth it at this point.

**Comment by edufford** [comment]
Actually, I just realized that switching the elif condition order would make it so that PID braking would not be able to activate whenever `is_decelerating` is False, such as for a speed overshoot while driving.  Maybe it would be better to try the error threshold min guard method instead.

File: ros/src/twist_controller/twist_controller_mod.py [hunk 1]
```diff
@@ -102,13 +104,21 @@ def control(self, target_linear_velocity, target_angular_velocity, current_linea
             control_mode = "Manual braking"
         elif velocity_error < -1 * limit_constant * current_linear_velocity:
```
Comment (edufford) [inline ros/src/twist_controller/twist_controller_mod.py]: Changing this line to:
`elif velocity_error < -1 * max(limit_constant * current_linear_velocity, 0.1):`
prevents the error threshold from getting too small near zero and seems to trigger the manual brake filter reset at each standing start.

Comment (dinoboy197) [reply ros/src/twist_controller/twist_controller_mod.py]: Great. Let me patch this in tomorrow and give it a try. Thanks!

Comment (dinoboy197) [reply ros/src/twist_controller/twist_controller_mod.py]: Tested and works well.

**Pull request merged**